misc/semver-checker/modules/semver-checker-core/src/test/java/org/ballerinalang/semver/checker/comparator/TypeComparatorTest.java

```diff
@@ -0,0 +1,527 @@
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.ballerinalang.semver.checker.comparator;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+import org.ballerinalang.semver.checker.exception.SemverTestException;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.FileReader;
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.util.LinkedList;
+import java.util.List;
+
+import static org.ballerinalang.semver.checker.util.TestUtils.executeTestData;
+
+/**
+ * Type definition comparator related test cases.
+ *
+ * @since 2201.2.0
+ */
+public class TypeComparatorTest {
+
+    private static final String TEST_ROOT = "src/test/resources/testcases/typeDefinition/";
+    private static final String TYPE_DESCRIPTOR_TEST_ROOT = TEST_ROOT + "typeDescriptor/";
+    private static final String BEHAVIOURAL_TYPE_DESCRIPTOR_TEST_ROOT = TYPE_DESCRIPTOR_TEST_ROOT + "behaviouralTypeDescriptor/";
+    private static final String SEQUENCE_TYPE_DESCRIPTOR_TEST_ROOT = TYPE_DESCRIPTOR_TEST_ROOT + "sequenceTypeDescriptor/";
+    private static final String SIMPLE_TYPE_DESCRIPTOR_TEST_ROOT = TYPE_DESCRIPTOR_TEST_ROOT + "simpleTypeDescriptor/";
+    private static final String STRUCTURAL_TYPE_DESCRIPTOR_TEST_ROOT = TYPE_DESCRIPTOR_TEST_ROOT + "structuredTypeDescriptor/";
+    private static final String OTHER_TYPE_DESCRIPTOR_TEST_ROOT = TYPE_DESCRIPTOR_TEST_ROOT + "otherTypeDescriptor/";
+    private static final String BEHAVIOURAL_FUNCTION_TYPE_DESCRIPTOR_TEST_ROOT = BEHAVIOURAL_TYPE_DESCRIPTOR_TEST_ROOT + "function/";
+    private static final String BEHAVIOURAL_OBJECT_TYPE_DESCRIPTOR_TEST_ROOT = BEHAVIOURAL_TYPE_DESCRIPTOR_TEST_ROOT + "object/";
+    private static final String BEHAVIOURAL_OBJECT_METHOD_DEFINITION_TYPE_DESCRIPTOR_TEST_ROOT = BEHAVIOURAL_OBJECT_TYPE_DESCRIPTOR_TEST_ROOT + "methodDefinition/";
```

Comment:
Shall we fix potential checkstyle violations in here? (line lengths should be less than 120 characters)

Reply:
+1